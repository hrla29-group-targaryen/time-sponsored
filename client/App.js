import React from 'react';
import axios from 'axios';

class App extends React.Component{
	constructor(props){
		super(props)
		this.state={
			data : null,
			counter: 0
		}
		this.increment=this.increment.bind(this)
	}

	// {
	//   "userId": 1,
	//   "id": 1,
	//   "title": "delectus aut autem",
	//   "completed": false
	// }

	componentDidMount(){
		axios('https://jsonplaceholder.typicode.com/todos/1')
		.then(({data}) => this.setState({data}))
		.catch(e=> console.log(e))
	}

	increment(){
		let curr = this.state.counter
		this.setState({counter: curr+1})
	}

	render(){
		// console.log(this.state.counter)
		return(
			<section>
				<h1>Hello</h1>
				<button id="counter_button" onClick={this.increment}>Click</button>
			{
				(this.state.data) &&
				(
					<div>
					id : {this.state.data.userId}
					</div>
				)
			}
			</section>
			)
	}
}

export default App;