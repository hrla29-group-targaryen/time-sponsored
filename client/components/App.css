.order-container{
	max-width: 75%;
    background: white;
    padding: 15px;
    min-height: 42px;
    display: flex;
    text-align: center;
    justify-content: center;
}

.order-change{
	background: transparent;
    border: none;
    color: #0058b8;
    font-size: 14px;	
    cursor: pointer;
}

.order-status{
	align-items: center;
    display: grid;
    text-align: left;	
    padding-right: 40px;
}

.order-status p{
	margin: 0;
    font-size: 14px;
}

.order-no-minimum{
	font-size: 12px !important;
	color: #13aa37;
}


.app-sponsor-container{
    max-width: 75%;
    background: white;
    padding: 15px;
}