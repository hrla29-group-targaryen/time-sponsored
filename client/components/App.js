import React from 'react';
import './App.css';
import Modal from './Modal/Modal';
import Order from './Order/Order';

import axios from 'axios';

class App extends React.Component{
	constructor(props){
		super(props)
		this.state= {
			isModalOpen : false,
			data: null
		}
		this.getRestaurantData = this.getRestaurantData.bind(this)
		this.toggleModal = this.toggleModal.bind(this)

	}

	componentDidMount(){
	  	this.getRestaurantData()
	}

	getRestaurantData(){
		const id = window.location.href.split("id=")[1]
	  	axios.get(`http://localhost:3000/api/data/${id}`)
	  	.then(({data}) => this.setState({data}))
	  	.catch(e=>console.log(e))
	}

	toggleModal(){
		this.setState({isModalOpen: !this.state.isModalOpen})
	}

	render(){
		const {isModalOpen} = this.state
		return(
			<section>
				<section className="order-container">
					<section className="order-status">
						<p>Delivery, ASAP {(this.state.data) && <span>{`(${this.state.data.waitingtime}m)`}</span>}</p>
						<p className="order-no-minimum">No minimum</p>
					</section>
					<button className="order-change" onClick={this.toggleModal}>Change</button> <br />
				</section>
				<section>
					{
						(isModalOpen && this.state.data) &&
						<Modal>
							<Order data={this.state.data} isModalOpen={isModalOpen} toggleModal={this.toggleModal}/>
						</Modal>
					}
				</section>
			</section>
			)
	}
}


export default App;