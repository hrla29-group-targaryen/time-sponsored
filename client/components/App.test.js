import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import App from './App';

describe('testing App component', ()=> {

	it('should return a single-node wrapper.', ()=> {
		expect(shallow(<App />).length).toEqual(1)
	})

	it('should grab a snapshot of the component.', ()=> {
		const component = renderer.create(<App />)
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	})

})
