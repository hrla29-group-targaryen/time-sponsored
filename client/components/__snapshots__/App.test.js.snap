// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing App component should grab a snapshot of the component. 1`] = `
<section>
  <section
    className="order-container"
  >
    <section
      className="order-status"
    >
      <p>
        Delivery, ASAP (35-45m)
      </p>
      <p
        className="order-no-minimum"
      >
        No minimum
      </p>
    </section>
    <button
      className="order-change"
      onClick={[Function]}
    >
      Change
    </button>
     
    <br />
  </section>
</section>
`;
